### ISSUE TITLE
RFC: change signature of hmset

### ISSUE THREAD

**Issue by lukepalmer** [Leonidas-from-XIV/orewa#17]
It seems more natural to have the signature be:

val hset
  :  t ->
  elements:(string * string) list ->
  string ->
  (int, [> common_error]) Deferred.Result.t

It seems odd to have ~element and ?elements, which I mostly need to write special case code each time to use. 

Happy to make the change, but I don't have a sense of how big of a deal it is to break compatibility, though. Let me know what you think.

**Comment by Leonidas-from-XIV** [comment]
I don't think it will be an issue with compatibility, since I don't think `hset` support is part of any release at the moment. I don't have a strong feeling about it, if you consider that a better API I am happy to apply that change.
